client/state/products-list/reducer.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	PRODUCTS_LIST_RECEIVE,
+	PRODUCTS_LIST_REQUEST,
+	PRODUCTS_LIST_REQUEST_FAILURE,
+	DESERIALIZE,
+	SERIALIZE,
+} from 'state/action-types';
+import { productsListSchema } from './schema';
+import { isValidStateWithSchema } from 'state/utils';
+
+/**
+ * Stores the complete list of products, indexed by the product key
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function items( state = null, action ) {
+	switch ( action.type ) {
+		case PRODUCTS_LIST_RECEIVE:
+			return action.productsList;
+
+		case SERIALIZE:
+			return state;
+
+		case DESERIALIZE:
+			if ( state !== null && isValidStateWithSchema( state, productsListSchema ) ) {
+				return state;
+			}
+			return null;
+	}
+
+	return state;
+}
+
+/**
+ * Tracks sites fetching state
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function isFetching( state = false, action ) {
```

Comment:
And to avoid persisting state, we don't pass in a schema:
```javascript
export const items = createReducer( null, {
    [ PRODUCTS_LIST_REQUEST ]: () => true,
    [ PRODUCTS_LIST_RECEIVE ]: () => false,
    [ PRODUCTS_LIST_REQUEST_FAILURE ]: () => false
} );
```